File: tests/qualification/language-wave/swift/Module.swift
import Vapor

public protocol Store { func save(_ value: String) async throws }
public struct UserStore: Store {
    public init() {}
    public func save(_ value: String) async throws { _ = value.count }
}
public extension UserStore {
    func route(_ app: Application) {
        app.get("users", use: listUsers)
    }
    private func listUsers(_ request: Request) async throws -> String { "ok" }
}

struct AmbiguousA { func same() {} }
struct AmbiguousB { func same() {} }
func repeated(_ store: UserStore) {
    _ = store.save
    _ = store.save
}

File: tests/qualification/language-wave/scala/Module.scala
package wave

trait Store { def save(value: String): Unit }
final case class User(name: String)
final class UserStore extends Store {
  override def save(value: String): Unit = println(value)
  def route(): Unit = save("users")
}
object UserStore { def apply(): UserStore = new UserStore() }
given ordering: Ordering[User] with
  def compare(left: User, right: User): Int = left.name.compareTo(right.name)
extension (store: UserStore) def repeated(): Unit = { store.save("a"); store.save("b") }
